lib/pure/uri.nim

```diff
@@ -289,6 +294,33 @@ func parseUri*(uri: string, result: var Uri) =
     assert res.scheme == "https"
     assert res.hostname == "nim-lang.org"
     assert res.path == "/docs/manual.html"
+
+    res = initUri()
```

Comment:
These tests are too exhaustive for "runnable examples". They should go into the relevant test file (IIRC it's called turi.nim). Please move it there.

You can add a simpler `strict` example in here, but keep it as succinct as possible, readers of documentation don't want to sift through many lines of code to figure out semantics.

Reply:
Makes sense! I simplified it a bit and moved tests to `turi.nim`.